src/proj.rs

```diff
@@ -255,6 +297,52 @@ impl Proj {
         }
     }
 
+    /// Return [Information](https://proj.org/development/reference/datatypes.html#c.PJ_INFO) about the current PROJ context
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn info(&self) -> Result<Projinfo, ProjError> {
+        let pinfo: PJ_INFO = unsafe { proj_info() };
+        Ok(Projinfo {
+            major: pinfo.major,
+            minor: pinfo.minor,
+            patch: pinfo.patch,
+            release: _string(pinfo.release)?,
+            version: _string(pinfo.version)?,
+            searchpath: _string(pinfo.searchpath)?,
+        })
+    }
+
+    /// Add a [resource file search path](https://proj.org/resource_files.html), maintaining existing entries.
+    ///
+    /// Changes to the search path [_should be_](https://github.com/OSGeo/PROJ/issues/2266) reflected in this
+    /// and **all** subsequently-created `Proj` instances, but **not** in other concurrently-existing instances.
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn set_search_paths<P: AsRef<Path>>(&self, newpath: P) -> Result<(), ProjError> {
+        let existing = self.info()?.searchpath;
+        let pathsep = if cfg!(windows) { ";" } else { ":" };
+        let mut individual: Vec<&str> = existing.split(pathsep).collect();
+        let np = Path::new(newpath.as_ref());
+        individual.push(np.to_str().ok_or(ProjError::Path)?);
+        let newlength = individual.len() as i32;
+        // convert path entries to CString
+        let paths_c = individual
+            .iter()
+            .map(|str| CString::new(*str))
+            .collect::<Result<Vec<_>, std::ffi::NulError>>()?;
+        // …then to raw pointers
+        let paths_p: Vec<_> = paths_c.iter().map(|cstr| cstr.as_ptr()).collect();
+        // …then pass the slice of raw pointers as a raw pointer (const char* const*)
+        // We pass a null pointer as the context, as we want the search path to be
+        // available to all contexts
+        let dctx: *mut PJ_CONTEXT = ptr::null_mut();
```

Comment:
I'm not at all familiar with this crate - this PR is the first I've looked into it, so I could be off base, but I'm confused at this point...

Here's my understanding of PJ_CONTEXT:

In the c interface, all proj config and state is wrapped up into a proj context. 

You can create and manage a context, and pass it into the methods which get and set state. As a convenience, you can use the global default context by passing in a null ptr instead of your own context.

Before this PR, this method seemed to be the only place we use the default context. I'm not sure why we were doing that - my first impulse is that it's strange. It seems surprising to me that a method which is on an explicit instance would also affect another instance (the global default context). 

The new network methods added in this PR `network_enabled`, `enable_network(enable: bool) -> Result<(), ProjError>`, etc. now also operate on the default ctx. Why is that? 

I would have expected that they would be impl methods on an instance of Proj, which read and write only to their Proj context and not touch the global default context.

```
let my_proj = Proj::new(a,b,c);
my_proj.get_url_endpoint()
```

If we did need/want a way to manipulate the global default context, maybe best to do that through a static instance like 
```rust
Proj::default.get_url_endpoint()
```

But take that critique with a grain of salt - I've never used this lib directly. 🙃

Reply:
Great, thank you for this!  
The issue here is one of my own making:
In order to create transformation / projection instances using the crate, you can call `Proj::new` / `Proj::new_known_crs`. Note that both of these methods first create a `ctx` – the implicit context creation was a design decision on my part because I thought it would be tedious to explicitly create e.g. a `Context` instance, _then_ call a projection / transformation method on it.  
However, modifying search paths, network download functionality etc requires a `ctx`, as you point out – changing network download and search paths has to be done _before_ the transformation / projection object is created, because their creation may require search path lookup or remote grid file download. But because there is now no access to a `ctx` independently of a `Proj` instance, these properties can't be modified for an existing instance: they have to be set before creation. So there are three choices:

1. Break the API by requiring a two-step process: context creation followed by transformation / projection specification which consumes the ctx, with modification to search paths and network functionality as an intermediate step;
2. Break the API by adding new arguments to `Proj::new` / Proj::new_known_crs`
3. Leave the API as-is currently, which is at odds to the `libproj` API – my feeling is that crate users don't generally want or need to control these properties for concurrently existing `Proj` instances, and the existing effects of modifying properties are explicitly detailed in the comments.